### PR TITLE
Logic for non-default BLS profiling

### PR DESCRIPTION
**Pull request by nv-braf**
Includes unit testing and I've run this on FaceDetect and confirmed it is working.

### CODE REVIEW

File: model_analyzer/config/generate/quick_run_config_generator.py [hunk 1]
```diff
@@ -321,6 +321,17 @@ def _get_next_model_run_config(
         model_run_config = ModelRunConfig(model_name, model_config,
                                           perf_analyzer_config)
 
+        bls_composing_model_configs = []
+        for bls_composing_model in self._bls_composing_models:
+            bls_composing_model_config = self._get_next_model_config(
+                bls_composing_model, model_index)
+            model_index += 1
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/quick_run_config_generator.py:328]: This model index code seems very fragile. A lot of baked in knowledge to know the implicit locations of the models. Let's make sure that is part of the refactor.

File: tests/test_quick_run_config_generator.py [hunk 2]
```diff
@@ -911,6 +930,138 @@ def _get_next_run_config_ensemble(self,
 
         self.assertEqual(perf_config['batch-size'], 1)
 
+    def _get_next_run_config_bls(self, max_concurrency=0, min_concurrency=0):
+        """
+        Test that get_next_run_config() creates a proper RunConfig for BLS
+
+        Sets up a case where the coordinate is [1,2,3,4,5,6], which corresponds to
```
Comment (tgerdesnv) [inline tests/test_quick_run_config_generator.py:937]: Is this comment correct? Seems like way more than 6 values below

File: tests/test_quick_run_config_generator.py [hunk 3]
```diff
@@ -911,6 +930,138 @@ def _get_next_run_config_ensemble(self,
...
+          - composing model 1 concurrency = 8*5*2 = 80
+          - composing model 2 max_batch_size = 32
+          - composing model 2 instance_count = 7
+          - composing model 2 concurrency = 32*7*2 = 448
+          - model concurrency = 12 (minimum value of [12, 80, 448])
+
+        Also,
+        - sequence batching should be on for model 1
+        - dynamic batching should be on for model 2
+        - existing values from the base model config should persist if they aren't overwritten
```
Comment (tgerdesnv) [inline tests/test_quick_run_config_generator.py:952]: Are these two statements tested below?

File: tests/test_quick_run_config_generator.py [hunk 4]
```diff
@@ -911,6 +930,138 @@ def _get_next_run_config_ensemble(self,
 
         self.assertEqual(perf_config['batch-size'], 1)
 
+    def _get_next_run_config_bls(self, max_concurrency=0, min_concurrency=0):
+        """
+        Test that get_next_run_config() creates a proper RunConfig for BLS
```
Comment (tgerdesnv) [inline tests/test_quick_run_config_generator.py:935]: I love this test if it does everything it says it does! Very thorough

File: tests/test_quick_run_config_generator.py (hunk 3 above)
Comment (nv-braf) [reply tests/test_quick_run_config_generator.py:952]: Yes, these are the lines that check it:
``` 
self.assertEqual(composing_model_config0.to_dict(),
                         expected_model_config0)
self.assertEqual(composing_model_config1.to_dict(),
                         expected_model_config1)
```

File: tests/test_quick_run_config_generator.py (hunk 4 above)
Comment (nv-braf) [reply tests/test_quick_run_config_generator.py:935]: It does! And, thanks.

File: tests/test_quick_run_config_generator.py (hunk 2 above)
Comment (nv-braf) [reply tests/test_quick_run_config_generator.py:937]: Reordered

File: model_analyzer/config/generate/quick_run_config_generator.py (hunk 1 above)
Comment (tgerdesnv) [reply model_analyzer/config/generate/quick_run_config_generator.py:328]: ok, we can revisit it when refactoring

File: tests/test_quick_run_config_generator.py (hunk 3 above)
Comment (nv-braf) [reply tests/test_quick_run_config_generator.py:954]: Good catch. Base model config check was missing. I've added it. Relevant PA config settings are checked (concurrency/batch size) here:
```
if max_concurrency:
            self.assertEqual(perf_config['concurrency-range'], max_concurrency)
        elif min_concurrency:
            self.assertEqual(perf_config['concurrency-range'], min_concurrency)
        else:
            self.assertEqual(perf_config['concurrency-range'], 12)

        self.assertEqual(perf_config['batch-size'], 1)
```

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**